### PR TITLE
Fix GCMap issue for read only unresolved field resolution and resolve virtual call sequence

### PR DESCRIPTION
**Pull request by r30shah**
For the read only unresolved field resolution and resolution of virtual , JIT helper glue code will have set up the return address to be the start of the resolution sequence where after returning from the helper, we again load the correct index or
address in case of field resolution and vTable offset for virtual call from the resolution data block. If the helper we call ends up triggering GC, the stack walker will use the JIT return address we passed to walk current JIT frame and find the GC map associated with the return address. The GC map logic subtracts 1 which means that unless we create the GC Map on an instruction before the return label, we will retrieve the wrong GC map. Correct fix here would be passing the address of next instruction after the snippet call in mainline as jitEIP when we call JIT helper. This commit fixes this issue.

Signed-off-by: Rahil Shah <rahil@ca.ibm.com>

### CODE REVIEW

**Comment by r30shah** [comment]
@fjeremic  Can I get a review on this changes?

**Comment by r30shah** [comment]
@fjeremic  This one is ready for review

File: runtime/compiler/z/codegen/J9UnresolvedDataReadOnlySnippet.cpp [hunk 1]
```diff
@@ -122,12 +123,11 @@ J9::Z::UnresolvedDataReadOnlySnippet::emitSnippetBody()
       // Branch to doneLabel (BRCL 0xF,[doneLabel])
       *reinterpret_cast<int16_t*>(cursor) = 0xC0F4;
 
-      addressA = reinterpret_cast<intptr_t>(doneLabel->getCodeLocation());
-      addressB = reinterpret_cast<intptr_t>(cursor);
+		intptr_t doneLabelAddress = reinterpret_cast<intptr_t>(doneLabel->getCodeLocation());
```
Comment (fjeremic) [inline runtime/compiler/z/codegen/J9UnresolvedDataReadOnlySnippet.cpp:126]: Tabs should be spaces here for consistency.

File: runtime/compiler/z/codegen/J9UnresolvedDataReadOnlySnippet.cpp [hunk 2]
```diff
@@ -67,12 +67,14 @@ J9::Z::UnresolvedDataReadOnlySnippet::UnresolvedDataReadOnlySnippet(
       intptr_t resolveDataAddress,
       TR::LabelSymbol *startResolveSequenceLabel,
       TR::LabelSymbol *volatileFenceLabel,
+      TR::LabelSymbol *snippetCallNextInstrLabel,
```
Comment (fjeremic) [inline runtime/compiler/z/codegen/J9UnresolvedDataReadOnlySnippet.cpp:70]: How about calling it what is is, i.e. `returnLabel`?

File: runtime/compiler/z/codegen/J9UnresolvedDataReadOnlySnippet.cpp [hunk 3]
```diff
@@ -89,26 +91,25 @@ J9::Z::UnresolvedDataReadOnlySnippet::emitSnippetBody()
    TR::GlobalFunctionCallData dataDestination(glueSymRef, getNode(), cursor, cg(), self());
    cursor = cg()->getObjFmt()->encodeGlobalFunctionCall(dataDestination);
 
-   // Relative address to the CCUnresolvedData*
```
Comment (fjeremic) [inline runtime/compiler/z/codegen/J9UnresolvedDataReadOnlySnippet.cpp:92]: We should probably leave this comment in.

Comment (r30shah) [reply runtime/compiler/z/codegen/J9UnresolvedDataReadOnlySnippet.cpp:92]: Added the comment back in https://github.com/eclipse/openj9/compare/ee40f0a3614cab3847b46a4b61c6fb2795d4b775..3ac390dbe99682be85f71980c4568d19c47c35dd

File: runtime/compiler/z/codegen/J9UnresolvedDataReadOnlySnippet.cpp [hunk 4]
```diff
@@ -89,26 +91,25 @@ J9::Z::UnresolvedDataReadOnlySnippet::emitSnippetBody()
...
-   TR_ASSERT_FATAL(cg()->canUseRelativeLongInstructions(addressA), "startResolveSequenceLabel [%p] is outside relative immediate range", addressA);
+   intptr_t startOfResolutionSeqLabelAddr = reinterpret_cast<intptr_t>(startResolveSequenceLabel->getCodeLocation());
+   TR_ASSERT_FATAL(cg()->canUseRelativeLongInstructions(startOfResolutionSeqLabelAddr), "startResolveSequenceLabel [%p] is outside relative immediate range", startOfResolutionSeqLabelAddr);
+   *reinterpret_cast<int32_t*>(cursor) = static_cast<int32_t>(startOfResolutionSeqLabelAddr - (intptr_t)helperCallRA);
+   cursor += 4;
 
-   *reinterpret_cast<int32_t*>(cursor) = static_cast<int32_t>(addressA - addressB);
+   intptr_t snippetCallNextInstrLabelAddress = reinterpret_cast<intptr_t>(snippetCallNextInstrLabel->getCodeLocation());
+   *reinterpret_cast<int32_t*>(cursor) = static_cast<int32_t>(snippetCallNextInstrLabelAddress - (intptr_t)helperCallRA);
    cursor += 4;
```
[the remark this answers is not in the document]
Comment (r30shah) [reply runtime/compiler/z/codegen/J9UnresolvedDataReadOnlySnippet.cpp:114]: Added the comment in https://github.com/eclipse/openj9/compare/3ac390dbe99682be85f71980c4568d19c47c35dd..602be0121b2e925ba5801a153abb585fe9008379

File: runtime/compiler/z/codegen/J9UnresolvedDataReadOnlySnippet.cpp (hunk 1 above)
Comment (r30shah) [reply runtime/compiler/z/codegen/J9UnresolvedDataReadOnlySnippet.cpp]: Fixed in https://github.com/eclipse/openj9/compare/ee40f0a3614cab3847b46a4b61c6fb2795d4b775..3ac390dbe99682be85f71980c4568d19c47c35dd

File: runtime/compiler/z/codegen/J9UnresolvedDataReadOnlySnippet.cpp (hunk 2 above)
Comment (r30shah) [reply runtime/compiler/z/codegen/J9UnresolvedDataReadOnlySnippet.cpp:70]: I chose this name it represents the nextInstruction after snippet call for GC, and in none of the path we actually return to this point.

**Review by fjeremic — APPROVED** [review]

**Pull request merged**